src/Services/SodiumService.php

```
@@ -131,4 +131,20 @@ protected function checkKey(string $key = null): string
 
         throw new KeyNotFoundException(KeyNotFoundException::NEITHER_KEY_NOT_FOUND_MESSAGE);
     }
+
+    /**
+     * Decrypt the value using a nonce.
+     *
+     * @param string $value the value to decrypt
+     * @param string $nonce The base64 encoded nonce;
+     */
+    public function decryptValueByNonce(string $value, string $nonce): string
+    {
+        try {
+            return $this->decrypt($nonce ? implode('.', [$nonce, $value]) : $value);
+        } catch (Exception $e) {
```

A little concern about the exception been swallowed silently. Since this the 3rd party library, the logic like this is not visible to the codebase. It's hard to determine the outcome of the decryption, is the decrypt successful that a clear text is deciphered, or the decryption failed that the original text is returned?